Updating index.js to match Node v4.3 format

mcloide opened the pull request:
This merge request will update the callbacks for `succeed` and `fail` based on AWS documentation -  http://docs.aws.amazon.com/lambda/latest/dg/nodejs-prog-model-using-old-runtime.html#transition-to-new-nodejs-runtime - aligning with the change for runtime to be Node v4.3.

This comes as a request from AWS on deprecation of Node v0.10 - https://forums.aws.amazon.com/ann.jspa?annID=4345

What it does:

context.succeed is replaced by

```
context.callbackWaitsForEmptyEventLoop = false;
callback(null, 'Success message');
```

context.fail is replaced by 

```
context.callbackWaitsForEmptyEventLoop = false;
callback(err, 'Fail message');
```

Also replaced the if statements for a switch to improve readability.

**Note** this was tested on a real aws lambda before publishing this PR. No errors reported by the lambda.

Commit:
Updating index.js to use Node v4.3 

As mentioned in AWS documentation to migrate from Node v0.10 to v4.3, context functions would need to be updated to use the new nomenclature.

ref: http://docs.aws.amazon.com/lambda/latest/dg/nodejs-prog-model-using-old-runtime.html#transition-to-new-nodejs-runtime

This commit will update the `context` functions of:

* context.fail(object)
* context.succeed()

To Node v4.3 format following what is suggested by the documentation provided by AWS.

Commit:
Correcting the handler parameter call.

mcloide commented:
@michaelwittig code updated.

michaelwittig commented on index.js:
```diff
@@ -69,10 +71,11 @@ exports.handler = function(event, context) {
+				context.callbackWaitsForEmptyEventLoop = false;
 				if (err) {
-					context.fail(err);
+					callback(err, 'Failed result');
```
usually, if the first parameter is set (an error), the second one is not passed.

mcloide replied:
👍

michaelwittig commented on index.js:
```diff
@@ -69,10 +71,11 @@ exports.handler = function(event, context) {
+					callback(err, 'Failed result');
 				} else {
-					context.succeed();
+					callback(null, 'Success message');
```
is a 2nd parameter needed? It will change the behaviour of the program.

mcloide replied:
No. It was suggested by AWS. Basically the callback will report anything as success as long it doesn't pass an Error object. I will remove and update.

michaelwittig reviewed (changes requested):
I just added two small comments. Let mw know what you think.

Commit:
Removing custom messages

To maintain compatibility with the original concept and application, removing the custom messages that was added.

mcloide commented:
Updated as requested.

michaelwittig commented:
thanks @mcloide !

mcloide commented:
@michaelwittig no problem. Happy to contribute.
